confluent_kafka/src/confluent_kafka.c

```diff
@@ -1245,6 +1260,7 @@ rd_kafka_conf_t *common_conf_setup (rd_kafka_type_t ktype,
 						"as a callable function");
 				rd_kafka_topic_conf_destroy(tconf);
 				rd_kafka_conf_destroy(conf);
+                                Py_XDECREF(ks8);
```

Comment:
I know it becomes inconsistent with the way `ks` is handled, but couldn't we just `Py_XDECREF` immediately so we only have it once and don't have to scatter it on all of these paths? In fact, it looks like we could do that for `ks` in some of the cases that have been converted here.

Reply:
On  Python 2.x ks8 will point to another object than ks, namely the utf8 representation of ks, and since `k` is a char pointer to ks's underlying memory we must retain the reference until all usage of `k` is over.

Ony Python 3 ks8 will always be NULL, thus causing XDECREF to be a no-op.

Reply:
ack, makes sense. Kinda sucks that the API naming doesn't make it really obvious when you're getting internal data vs a completely new object.